### PR TITLE
Fix: forget the user’s consents for analytics on logout

### PR DESCRIPTION
**Pull request by nimau**
This PR reset the user's consents for analytics on logout (as requested in #106)

### CODE REVIEW

File: ElementX/Sources/Application/AppSettings.swift [hunk 1]
```diff
@@ -128,6 +128,12 @@ final class AppSettings: ObservableObject {
     /// `true` when the user has opted in to send analytics.
     @UserSetting(key: UserDefaultsKeys.enableAnalytics.rawValue, defaultValue: false, persistIn: store)
     var enableAnalytics
+    
+    /// Reset user consents for analytics
+    static func forgetAnalyticsConsents() {
```
Comment (stefanceriu) [inline ElementX/Sources/Application/AppSettings.swift:133]: @pixlwave and I were chatting about this and we believe we have a better solution:
- change `enableAnalytics` to be an enum with 3 values stored as a RawRepresentable with a default value of `.unknown` or similar
- remove the `forgetAnalyticsConsents` and just use `ServiceLocator.shared.settings.enableAnalytics  = .unknown` when you want to reset it

**Review by stefanceriu — CHANGES_REQUESTED** [review]

File: ElementX/Sources/Services/Analytics/AnalyticsConsentState.swift [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+enum AnalyticsConsentState: String {
+    case optOut
+    case optIn
```
Comment (stefanceriu) [inline ElementX/Sources/Services/Analytics/AnalyticsConsentState.swift:21]: Should these be called `optedIn` and `optedOut` maybe?

File: ElementX/Sources/Application/AppSettings.swift (hunk 1 above)
Comment (nimau) [reply ElementX/Sources/Application/AppSettings.swift]: Hi @stefanceriu, I've updated the PR to reflect this change.
Also I took the liberty of renaming the UserDefaultKey from `enableAnalytics` to `analyticsConsentState` to better reflect the purpose of this key.

File: ElementX/Sources/Application/AppCoordinator.swift [hunk 3]
```diff
@@ -295,7 +295,8 @@ class AppCoordinator: AppCoordinatorProtocol {
             tearDownUserSession()
             
             // reset analytics
-            ServiceLocator.shared.analytics.reset()
+            ServiceLocator.shared.analytics.optOut()
+            ServiceLocator.shared.analytics.resetConsentState()
```
Comment (stefanceriu) [inline ElementX/Sources/Application/AppCoordinator.swift:299]: At this point wouldn't it be better to keep the `reset` method that would `optOut` + `resetConsent` internally?

**Review by stefanceriu — APPROVED** [review]
Looks great 👍

File: ElementX/Sources/Services/Analytics/AnalyticsConsentState.swift (hunk 2 above)
Comment (nimau) [reply ElementX/Sources/Services/Analytics/AnalyticsConsentState.swift]: Done.

File: ElementX/Sources/Application/AppCoordinator.swift (hunk 3 above)
Comment (nimau) [reply ElementX/Sources/Application/AppCoordinator.swift:299]: There is a `reset` method on `Analytics` but it is for resetting the analytics client and the bug report service. Do you mean to move these two calls inside a new private `reset` method inside the `AppCoordinator`?

Comment (stefanceriu) [reply ElementX/Sources/Application/AppCoordinator.swift:299]: No no, sorry, I didn't realise there's another reset method in there

**Pull request merged**